app/services/config_report_importer.rb

```diff
@@ -1,10 +1,18 @@
 class ConfigReportImporter < ReportImporter
+  def report_name_class
+    ConfigReport
+  end
+
   def self.authorized_smart_proxy_features
-    @authorized_smart_proxy_features ||= super + ['Puppet']
+    ReportImporter.authorized_smart_proxy_features
   end
 
-  def report_name_class
-    ConfigReport
+  def self.register_smart_proxy_feature(feature)
+    ReportImporter.register_smart_proxy_feature(feature)
+  end
+
+  def self.unregister_smart_proxy_feature(feature)
+    ReportImporter.unregister_smart_proxy_feature(feature)
   end
```

Comment:
Perhaps I'm missing something, but `ConfigReportImporter` inherits from `ReportImporter` so isn't this redundant?

Reply:
It is not actually, because the definition of the parent method means it's defined on class, so you'd define another array on `ConfigReportImporter` tho this issue would appear in any of the descendants. That is one of the reasons to not have registries in a parent classes.

Reply:
@lzap could we deprecate these then and disable it's usage through ConfigReportImporter?

Reply:
Ah right, that's why I dislike static methods with state.